docs/DEPRECATION.md

```diff
@@ -0,0 +1,83 @@
+# Deprecations in Vector
+
+This document covers Vector's deprecation policy and process.
+
+In the course of Vector's development it can be necessary to deprecate configuration and (rarely) features to keep
+Vector maintainable and its configuration interface consistent for users. To avoid breaking compatibility abruptly, we
+follow the following deprecation policy.
+
+## Policy
+
+Vector will retain deprecated configuration or features for at least one minor version (this will transition to one
+major version when Vector hits 1.0).
+
+This means that deprecations will be eligible for removal in the next minor release after they are announced; however,
+we will typically aim to support deprecations for a longer time period depending on their maintenance burden. For
```

Comment:
What does maintenance burden refer to? Is this just a deprecation with a migration that would be difficult/tedious to complete?

Reply:
We've generally been keeping deprecated options/features until "1.0", assuming keeping them "isn't a burden". When they are painful or important to remove we've done more of the announce during minor release, remove next release.

Reply:
Yeah, what Spencer said. I'll clarify it here. Basically we'd remove if it is causing us substantial pain to maintain. This is, though, weighed against the user impact you are describing. If it is hard to migrate, we might expend more effort supporting the deprecated behavior.